engine/class_modules/paladin/sc_paladin.cpp

```diff
@@ -1658,19 +1820,20 @@ void paladin_t::create_buffs()
 
   buffs.avengers_might = make_buff<stat_buff_t>( this, "avengers_might", find_spell( 272903 ) )
        -> add_stat( STAT_MASTERY_RATING, azerite.avengers_might.value() );
-
   buffs.seraphim = make_buff( this, "seraphim", talents.seraphim )
        -> add_invalidate( CACHE_CRIT_CHANCE )
        -> add_invalidate( CACHE_HASTE )
        -> add_invalidate( CACHE_MASTERY )
        -> add_invalidate( CACHE_VERSATILITY )
        -> set_cooldown( 0_ms ); // let the ability handle the cooldown
-
   buffs.holy_avenger = make_buff( this, "holy_avenger", talents.holy_avenger )
         -> set_cooldown( 0_ms ); // handled by the ability
-  buffs.blessing_of_dawn = make_buff( this, "blessing_of_dawn", legendary.from_dusk_till_dawn -> effectN( 1 ).trigger() );
-  buffs.blessing_of_dusk = make_buff( this, "blessing_of_dusk", legendary.from_dusk_till_dawn -> effectN( 2 ).trigger() )
-        -> set_default_value( legendary.from_dusk_till_dawn -> effectN( 2 ).trigger() -> effectN( 1 ).percent() );
+  buffs.devotion_aura = make_buff( this, "devotion_aura", find_class_spell( "Devotion Aura" ) );
```

Comment:
You could set the default value here and use it instead of checking the buff, then using its damage reduction value from looking at the data.